templates/leptos-ssr/content/src/server.rs

```diff
@@ -1,19 +1,43 @@
-use leptos_spin::{render_best_match_to_stream, RouteTable, server_fn::register_explicit};
-use spin_sdk::http::{IncomingRequest, ResponseOutparam};
-use spin_sdk::http_component;
+use std::sync::Arc;
 
-#[http_component]
-async fn handle_{{project-name | snake_case}}(req: IncomingRequest, resp_out: ResponseOutparam) {
-    let mut conf = leptos::get_configuration(None).await.unwrap();
-    conf.leptos_options.output_name = "{{project-name | snake_case}}".to_owned();
+use leptos::config::get_configuration;
+use leptos_wasi::prelude::{Executor, IncomingRequest, ResponseOutparam, WasiExecutor};
+use wasi::exports::http::incoming_handler::Guest;
+use wasi::http::proxy::export;
 
-    // Register server functions
-    register_explicit::<crate::app::SaveCount>();
+use crate::app::{shell, App, SaveCount};
 
-    let app = crate::app::App;
+struct LeptosServer;
 
-    let mut routes = RouteTable::build(app);
-    routes.add_server_fn_prefix("/api").unwrap();
+impl Guest for LeptosServer {
+    fn handle(request: IncomingRequest, response_out: ResponseOutparam) {
+        // Initiate a single-threaded [`Future`] Executor so we can run the
+        // rendering system and take advantage of bodies streaming.
+        let executor = WasiExecutor::new(leptos_wasi::executor::Mode::Stalled);
+        Executor::init_local_custom_executor(executor.clone())
+            .expect("cannot init future executor");
+        executor.run_until(async {
+            handle_request(request, response_out).await;
+        })
+    }
+}
+
+async fn handle_request(request: IncomingRequest, response_out: ResponseOutparam) {
+    use leptos_wasi::prelude::Handler;
 
-    render_best_match_to_stream(req, resp_out, &routes, app, &conf.leptos_options).await
+    let mut conf = get_configuration(None).unwrap();
+    conf.leptos_options.output_name = Arc::from("{{project-name | snake_case}}".to_owned());
+    let leptos_options = conf.leptos_options;
+
+    Handler::build(request, response_out)
+        .expect("could not create handler")
+        .with_server_fn::<SaveCount>()
```

Comment:
It's worth commenting that developers need to add new server functions here.  It's quite easy to miss now it's in the middle of a fluent chain.

Reply:
Good point, I will add this now